walnut/cmd/convert.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/AliceO2Group/Control/core/workflow"
+	"github.com/AliceO2Group/Control/walnut/converter"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io/ioutil"
+	"os"
+)
+
+// convertCmd represents the convert command
+var convertCmd = &cobra.Command{
+	Use:   "convert",
+	Short: "Converts a DPL Dump to the required formats.",
+	Long: `The convert command takes a DPL input and outputs task and workflow templates. Optional flags can be provided to
+specify which modules were used during dump generation. By default, both templates are produced. Control-OCCPlugin is 
```

Comment:
`s/were used during dump generation/should be used when generating task templates/`

What do you mean with "By default, both templates are produced"? Is there an option to only produce TTs or WFTs?

Reply:
My bad, I had a `--template` flag in a previous version which would allow the user to selectively generate either `task` or `workflow` templates. I removed that temporarily because I had some trouble with the flag value check. We can either make a JIRA issue for this to come back at a later date or re-implement it now. Either works for me.

Reply:
I don't think there's much of a use case for outputting only TTs or only WFTs. The output of a DPL dump should always be 1 WFT + an as-needed number of TTs.